task/build-image-index/0.1/build-image-index.yaml

```diff
@@ -79,10 +85,19 @@ spec:
       for i in $@
       do
         TOADD="$i"
-        if [[ $(echo $i | tr -cd ":" | wc -c) == 2 ]]; then
+        if [[ $(echo "$i" | tr -cd ":" | wc -c) == 2 ]]; then
           #we need to remove the tag, and just reference the digest
           #as tag + digest is not supported
-          TOADD="$(echo $i | cut -d: -f1)@sha256:$(echo $i | cut -d: -f3)"
+          TOADD_REPOSITORY="$(echo "$i" | cut -d: -f1)"
+          TOADD_DIGEST="sha256:$(echo "$i" | cut -d: -f3)"
+          TOADD="${TOADD_REPOSITORY}@${TOADD_DIGEST}"
+        fi
+        if [[ "$ALWAYS_BUILD_INDEX" != "true" ]]; then
+          echo "Skipping image index generation. Returning results for $TOADD"
+          echo -n "${TOADD_REPOSITORY}" > "$(results.IMAGE_URL.path)"
+          echo -n "${TOADD_DIGEST}" > "$(results.IMAGE_DIGEST.path)"
+          echo -n "${TOADD}" > "$(results.IMAGES.path)"
+          exit 0
```

Comment:
IMO `ALWAYS_BUILD_INDEX=false` should be ignored if there's multiple IMAGES

Reply:
Or we can have a `BUILD_INDEX=false` and raise an error if there's multiple IMAGES

Reply:
Sounds good. We can always revisit this in the future as loosening of the restriction won't be a breaking change but adding functionality.